pkg/dom0-ztools/rootfs/bin/eve

```diff
@@ -251,8 +251,20 @@ __EOT__
           echo "Updated memory-monitor configuration"
           ;;
  verbose) # first lets find our piping process
-          for PIPE in $(pgrep cat); do
-             [ "$(readlink /proc/"$PIPE"/fd/0)" = /run/diag.pipe ] && break
+          PIPE=""
+          while [ "$PIPE" = "" ]
+          do
+              for pid in $(pgrep cat)
```

Comment:
Are you trying to wait for the creation of the /run/diag.pipe file?  You can use inotifyd for that, something like this:

```sh
while read proc;
do
  if [ "$(echo $proc | grep /run/diag.pipe)" ]; then
	  echo "FILE CREATED!"
	  break
  fi
done < <(inotifyd - /run:n)
```

Reply:
No, I wait for the creation of the process that reads from it.